erpnext/selling/doctype/sales_order/sales_order_list.js

```diff
@@ -54,5 +54,341 @@ frappe.listview_settings['Sales Order'] = {
+					Are you sure you want to create {0} Production Plan(s)?`, [selected_docs.length]),
+				() => {
+					frappe.call({
+						method: "erpnext.selling.doctype.sales_order.sales_order.create_muliple_production_plans",
+						args: {
+							"orders": docnames
+						},
+						freeze: true,
+						callback: (r) => {
+							if (!r.exc) {
+								if (r.message.length === 0) {
+									return;
+								}
+
+								let message = "";
+
+								// loop through each created order and render linked Production Plans
+								let created_order_message = "";
+								let created_orders = r.message.filter(order => order.created === true);
+								for (let order of created_orders) {
+									let production_plans = order.production_plans
+										.map(production_plan => frappe.utils.get_form_link("Production Plan", production_plan, true))
+										.join(", ");
+
+									created_order_message += `<li><strong>${order.customer}</strong> (${order.sales_order}): ${production_plans}</li>`;
+								}
+
+								if (created_order_message) {
+									message += `The following Production Plan were created:<br><br><ul>${created_order_message}</ul>`;
+								}
+
+								frappe.msgprint(__(message));
+
+								// if validation messages are found, append at the bottom of our message
+								if (r._server_messages) {
+									let server_messages = JSON.parse(r._server_messages);
+									for (let server_message of server_messages) {
+										frappe.msgprint(__(JSON.parse(server_message).message));
+									}
+									// delete server messages to avoid Frappe eating up our msgprint
+									delete r._server_messages;
+								}
+
+								listview.refresh();
+							}
+						}
+					});
+				});
+			}
+		};
+
+		listview.page.add_actions_menu_item(__('Create Pick Lists'), create_pick_list_action, false);
+		listview.page.add_actions_menu_item(__('Create Production Plan'), create_production_plan_action, false);
+		listview.page.add_actions_menu_item(__('Create Sales Invoices'), create_sales_invoice_action, false);
+		listview.page.add_actions_menu_item(__('Create Delivery Note'), create_delivery_note_action, false);
+
+		const send_email_action = () => {
+			const selected_docs = listview.get_checked_items();
+			const doctype = listview.doctype;
+			if (selected_docs.length > 0) {
+				let title = selected_docs[0].title;
+				for (let doc of selected_docs) {
+					if (doc.docstatus !== 1) {
+						frappe.throw(__("Cannot Email Draft or cancelled documents"));
+					}
+					if (doc.title !== title) {
+						frappe.throw(__("Select only one customer's sales orders"));
+					}
+				}
+				frappe.call({
+					method: "bloomstack_core.utils.get_contact",
+					args: { "doctype": doctype, "name": selected_docs[0].name, "contact_field": "customer" },
+					callback: function (r) {
+						frappe.call({
+							method: "bloomstack_core.utils.get_document_links",
+							args: { "doctype": doctype, "docs": selected_docs },
```

Comment:
@nmimsnikhil here is a dependency with bloomstack module. move this stuff to erpnext too